### ISSUE TITLE
LineageOS 19 complains about GmsCore required libraries

### ISSUE THREAD

**Issue by waltercool** [lineageos4microg/android_vendor_partner_gms#13]
```
[  0% 12/44161] Verifying uses-libraries: vendor/partner_gms/GmsCore/GmsCore.apk
FAILED: out/target/common/obj/APPS/GmsCore_intermediates/enforce_uses_libraries.status
/bin/bash -c "(rm -f out/target/common/obj/APPS/GmsCore_intermediates/enforce_uses_libraries.status ) && (build/soong/scripts/manifest_check.py           --enforce-uses-libraries        --enforce-uses-libraries-status out/target/common/obj/APPS/GmsCore_intermediates/enforce_uses_libraries.status      --aapt out/host/linux-x86/bin/aapt      --uses-library com.android.location.provider            --dexpreopt-config out/target/product/beyond1lte/obj/JAVA_LIBRARIES/com.android.location.provider_intermediates/dexpreopt.config            vendor/partner_gms/GmsCore/GmsCore.apk )"
error: mismatch in the <uses-library> tags between the build system and the manifest:
        - required libraries in build system: [com.android.location.provider]
                         vs. in the manifest: [com.android.location.provider]
        - optional libraries in build system: []
                         vs. in the manifest: [androidx.window.extensions, androidx.window.sidecar]
        - tags in the manifest (vendor/partner_gms/GmsCore/GmsCore.apk):
                uses-library:'com.android.location.provider'            uses-library-not-required:'androidx.window.extensions'          uses-library-not-required:'androidx.window.sidecar'
note: the following options are available:
        - to temporarily disable the check on command line, rebuild with RELAX_USES_LIBRARY_CHECK=true (this will set compiler filter "verify" and disable AOT-compilation in dexpreopt)
        - to temporarily disable the check for the whole product, set PRODUCT_BROKEN_VERIFY_USES_LIBRARIES := true in the product makefiles
        - to fix the check, make build system properties coherent with the manifest
        - see build/make/Changes.md for details

```

**Comment by notmyrealhandle** [comment]
I just ran into this while  building microg for LineageOS 18.1.

I see this [PR](https://github.com/lineageos4microg/android_vendor_partner_gms/pull/12); however being impatient I manually applied the change.

Also ty @mikalackis for the tip about android.permission.MANAGE_USB.  I suspect that has saved me some grief.

**Comment by giovEra** [comment]
@bananer, it seems that the linage4microg build server is not affected by this error, in fact [several roms](https://download.lineage.microg.org/?sort=time&order=desc) (lineage-19.1) have been compiled since the new version of GmsCore was committed. 

Instead like the other participants in this discussion, I have had build failures due to the lack of the optional libraries in the Android.xml file that seem to be needed for the new version of GmsCore.

I would like to know if you have any idea why this bizarre behavior is happening. 🤔 

Thank you 😄

**Comment by GitteHuber** [comment]
Building 19 works fine for me, but building 18.1 fails with

`FAILED: ninja: 'out/target/common/obj/JAVA_LIBRARIES/androidx.window.extensions_intermediates/javalib.jar', needed by 'out/target/product/avicii/obj/APPS/GmsCore_intermediates/dexpreopt.zip', missing and no known rule to make it
`
I'm using the same prebuilts for both build attempts:
` <project name="lineageos4microg/android_vendor_partner_gms" path="vendor/partner_gms" remote="github" revision="master" />`

Are there any other dependencies that have to met (or repos to be added) in 18.1? Are there device specific differences concerning these new optional dependencies?

**Comment by petefoth** [comment]
> building 18.1 fails with
> 
> `FAILED: ninja: 'out/target/common/obj/JAVA_LIBRARIES/androidx.window.extensions_intermediates/javalib.jar', needed by 'out/target/product/avicii/obj/APPS/GmsCore_intermediates/dexpreopt.zip', missing and no known rule to make it ` I'm using the same prebuilts for both build attempts: ` <project name="lineageos4microg/android_vendor_partner_gms" path="vendor/partner_gms" remote="github" revision="master" />`

I'm getting this build error too when I try to build 18.1. Any ideas for fixes or workarounds?

**Comment by petefoth** [comment]
Your Android.mk is missing the two lines added in [this commit](https://github.com/lineageos4microg/android_vendor_partner_gms/commit/881962077efaa799cb879df19639497558b13f75) and [this commit](https://github.com/lineageos4microg/android_vendor_partner_gms/commit/f0fc4d689183860c0de19dd3461641a04a9b7cd7) to fix the issue with building 19.1
```
LOCAL_USES_LIBRARIES := com.android.location.provider
LOCAL_OPTIONAL_USES_LIBRARIES := androidx.window.extensions androidx.window.sidecar
```
So it may be that fixing the 19.1 builds has broke 19.1. I'll experiment and report back

**Comment by petefoth** [comment]
Yes, the fixes for LOS 19 broke the LOS 18 builds - see #17. That was fixed in #18 

Closing
